File: react-native/src/theta-repository/theta-state/capture-status.ts
/** Capture Status constants */
export const CaptureStatusEnum = {
  /** Undefined value */
  UNKNOWN: 'UNKNOWN',
  /** Performing continuously shoot */
  SHOOTING: 'SHOOTING',
  /** In standby */
  IDLE: 'IDLE',
  /** Self-timer is operating */
  SELF_TIMER_COUNTDOWN: 'SELF_TIMER_COUNTDOWN',
  /** Performing multi bracket shooting */
  BRACKET_SHOOTING: 'BRACKET_SHOOTING',
  /** Converting post file... */
  CONVERTING: 'CONVERTING',
  /** Performing timeShift shooting */
  TIME_SHIFT_SHOOTING: 'TIME_SHIFT_SHOOTING',
  /** Performing continuous shooting */
  CONTINUOUS_SHOOTING: 'CONTINUOUS_SHOOTING',
  /** Waiting for retrospective video... */
  RETROSPECTIVE_IMAGE_RECORDING: 'RETROSPECTIVE_IMAGE_RECORDING',
  /** Performing burst shooting */
  BURST_SHOOTING: 'BURST_SHOOTING',
  /**
   * In the case of time-lag shooting by manual lens,
   * set while waiting for the second shot to be taken after the first shot is completed.
   */
  TIME_SHIFT_SHOOTING_IDLE: 'TIME_SHIFT_SHOOTING_IDLE',
} as const;

/** type definition of CaptureStatusEnum */
export type CaptureStatusEnum =
  (typeof CaptureStatusEnum)[keyof typeof CaptureStatusEnum];

File: flutter/lib/state/capture_status.dart
/// Capture Status
enum CaptureStatusEnum {
  /// Undefined value
  unknown('UNKNOWN'),

  /// Capture status. Performing continuously shoot
  shooting('SHOOTING'),

  /// Capture status. In standby
  idle('IDLE'),

  /// Capture status. Self-timer is operating
  selfTimerCountdown('SELF_TIMER_COUNTDOWN'),

  /// Capture status. Performing multi bracket shooting
  bracketShooting('BRACKET_SHOOTING'),

  /// Capture status. Converting post file...
  converting('CONVERTING'),

  /// Capture status. Performing timeShift shooting
  timeShiftShooting('TIME_SHIFT_SHOOTING'),

  /// Capture status. Performing continuous shooting
  continuousShooting('CONTINUOUS_SHOOTING'),

  /// Capture status. Waiting for retrospective video...
  retrospectiveImageRecording('RETROSPECTIVE_IMAGE_RECORDING'),
  
  /// Capture status. Performing burst shooting
  burstShooting('BURST_SHOOTING'),
  ;

  final String rawValue;

  const CaptureStatusEnum(this.rawValue);

  @override
  String toString() {
    return rawValue;
  }

  static CaptureStatusEnum? getValue(String rawValue) {
    return CaptureStatusEnum.values.cast<CaptureStatusEnum?>().firstWhere(
        (element) => element?.rawValue == rawValue,
        orElse: () => null);
  }
}
